dollarskip.c

```diff
@@ -1,13 +1,25 @@
 #include <stdlib.h>
 #include <string.h>
-char command[8096] = {0};
 int main(int argc, char **argv)
 {
-for (int i=1; i<argc; i++){
-    strcat(strcat(command,argv[i])," ");
-    
-}
-command[strlen(command) - 1] = '\0';
-
-system(command);
+	char command[8096] = {0};
+	if(argc>1) {
+		char shell[10], *env=getenv("SHELL"); //get the env var 'SHELL'
```

Comment:
```suggestion
		char shell[128] = {0}, *env=getenv("SHELL"); //get the env var 'SHELL'
```

Reply:
there is no need to make the `shell` char larger, as I said it only holds the shell name (longest name I know of is 4 characters (`bash`, `dash`, `fish`)) + another 5 characters (` -c '`) that leaves one extra character for the end character.
@CleanMachine1 if you feel the margin is to small then maybe a size of 11 or 12 characters is better?

there is no need to initialize it (though it doesn't affect anything) because `strcpy` is used to copy `*env` to `shell` and it does that automatically (I think, at least I know it auto detects the start of the variable or something like that).

Reply:
True @Itai-Nelken but remember: `$SHELL` could be an absolute path like `/usr/bin/bash` or even someone's custom path like `/home/me/Downloads/foo-bar-baz-v1.2.3/build/Debug/fish`.

Reply:
```
echo $SHELL
/bin/bash
```

Reply:
You are right @bl-ue, I never thought about that...